modules/data.land/R/IC_BADM_Utilities.R

```diff
@@ -0,0 +1,296 @@
+        
+        unit.ready <- ifelse(unit.in == "gC m-2",
+                             "g/m^2",
+                             ifelse(unit.in == "kgDM m-2", "kg/m^2",
+                                    "kg/m^2"))
+        
+        if (length(unit.in) == 0)
+          unit.ready <- "kg/m^2"
+        
+        Date.in <- Gdf %>%
+          dplyr::filter(VARIABLE %>% grepl("DATE", .)) %>%
+          dplyr::pull(DATAVALUE) %>%
+          as.Date()
+        if (length(Date.in) == 0)
+          Date.in <- NA
+        #----------------collect
+        # if it's biomass
+        if (type == "*_BIOMASS") {
+          Oregan.in <- Gdf %>%
+            dplyr::filter(VARIABLE %>% grepl("ORGAN", .)) %>%
+            dplyr::pull(DATAVALUE)
+          
+          
+          PlantWoodIni <-
+            udunits2::ud.convert(Gdf$DATAVALUE[1],  unit.ready, "kg/m^2")#       "AG_BIOMASS_CROP","AG_BIOMASS_SHRUB","AG_BIOMASS_TREE","AG_BIOMASS_OTHER"
+          
+        } else if (type == "*SOIL") {
+          val <- Gdf %>%
+            dplyr::filter(VARIABLE %>% grepl("SOIL_STOCK_C_ORG", .)) %>% #"SOIL_STOCK_C_ORG"
+            dplyr::pull(DATAVALUE) %>%
+            as.numeric()
+          
+          if (length(val) > 0)
+            SoilIni <- udunits2::ud.convert(val,  "g/m^2", "kg/m^2")
+          
+        } else if (type == "*_LIT_BIOMASS") {
+          litterIni <-
+            udunits2::ud.convert(Gdf$DATAVALUE[1],  unit.ready, "kg/m^2")
+          
+        } else if (type == "*_ROOT_BIOMASS") {
+          Rootini <-
+            udunits2::ud.convert(Gdf$DATAVALUE[1],  unit.ready, "kg/m^2")
+          
+        }
+        return(
+          data.frame(
+            Site = Gdf$SITE_ID %>% unique(),
+            Var = Gdf$VARIABLE[1],
+            Date = Date.in,
+            Organ = Oregan.in,
+            PlantWIni = PlantWoodIni,
+            SoilIni = SoilIni,
+            LitterIni = litterIni,
+            RootIni = Rootini
+          )
+        )
+    })
+ #cleaning
+ind <- apply(entries[,5:8], 1, function(x) all(is.na(x)))
+entries <- entries[-which(ind),]
+
+
+
+  return(entries)
+}
+
+
+#' netcdf.writer.BADAM
+#'
+#' @param lat numeric latitude
+#' @param long numeric longitude
+#' @param siteid site id as a string
+#' @param outdir outputdir which you want to store the IC netcdf file
+#'
+#' @return a string of the address where the IC file is stored
+#' @export
+#'
+netcdf.writer.BADAM <- function(lat, long, siteid, outdir ){
+  #--
+  input <- list()
+  dims <- list(lat = lat ,
+               lon = long,
+               time = 1)
+
+  #Reading in the BADM data
+  entries <- Read.IC.info.BADM (lat, long)
+  
+  PWI <- entries$PlantWIni[!is.na(entries$PlantWIni)]
+  LIn <- entries$LitterIni[!is.na(entries$LitterIni)]
+  SIn <- entries$SoilIni[!is.na(entries$SoilIni)]
+  
+  
+  variables <- list(
+    SoilMoistFrac = 1,
+    SWE = 0
+  )
+  
+  if (length(PWI)>0 ) variables <- c(variables, wood_carbon_content = sample(PWI, 1, replace = T))
```

Comment:
Watch your units. kg m-2 and kg C m-2 are not the same thing. You earlier said the BADM variables were in kg m-2